client/landing/stepper/declarative-flow/internals/steps-repository/components/paid-memberships-control/index.tsx

```diff
@@ -0,0 +1,47 @@
+import { Onboard } from '@automattic/data-stores';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useI18n } from '@wordpress/react-i18n';
+import { ChangeEvent } from 'react';
+import FormInputCheckbox from 'calypso/components/forms/form-checkbox';
+import FormFieldset from 'calypso/components/forms/form-fieldset';
+import FormLabel from 'calypso/components/forms/form-label';
+import InfoPopover from 'calypso/components/info-popover';
+import { ONBOARD_STORE } from 'calypso/landing/stepper/stores';
+import type { OnboardSelect } from '@automattic/data-stores';
+import './style.scss';
+
+const PaidMembershipsControl = () => {
+	const { __ } = useI18n();
+	const paidSubscribers = useSelect(
+		( select ) => ( select( ONBOARD_STORE ) as OnboardSelect ).getPaidSubscribers(),
+		[]
+	);
+	const { setPaidSubscribers, setGoals, resetGoals } = useDispatch( ONBOARD_STORE );
```

Comment:
Note we are directly selecting and dispatching the value of paidSubscribers from the redux onboard store. 

We had been pulling in that value, storing it in local state (useState), and then updating the redux store on form submit. And we were doing all that in the newseltter-setup/index.tsx file.

That seems redundant, and doing it this way allows us to consolidate all the logic for this component within the component, rather than having logic spread out across the newsletter setup screen, newsletter post setup screen, and useSetupFormInitialValues hook.

Reply:
Yep agreed, we noted that earlier in development ourselves also but we were still pretty much learning how the entire signup framework works, and we merged some code before we knew we could use "goals" option.

Thanks for the cleanup!